tools/converter/mule2ballerina/src/main/java/org/wso2/ei/tools/mule2ballerina/configreader/ConfigReader.java

```diff
@@ -0,0 +1,224 @@
+import org.wso2.ei.tools.mule2ballerina.model.BaseObject;
+import org.wso2.ei.tools.mule2ballerina.model.Flow;
+import org.wso2.ei.tools.mule2ballerina.model.GlobalConfiguration;
+import org.wso2.ei.tools.mule2ballerina.model.Processor;
+import org.wso2.ei.tools.mule2ballerina.model.Root;
+import org.wso2.ei.tools.mule2ballerina.util.Constant;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.EndElement;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * {@code ConfigReader} class reads mule configuration file and builds the Mule tree
+ */
+public class ConfigReader {
+
+    private static Logger logger = LoggerFactory.getLogger(ConfigReader.class);
+
+    private ElementMapper mapperObject;
+    private AttributeMapper attributeMapper;
+    private Root mRoot;
+    private boolean flowStarted = false;
+    private List<String> unIdentifiedElements;
+
+    public ConfigReader() {
+        mapperObject = new ElementMapper();
+        attributeMapper = new AttributeMapper();
+        mRoot = new Root();
+        unIdentifiedElements = new ArrayList<String>();
+    }
+
+    public void readXML(InputStream inputStream) {
+
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+        try {
+            XMLEventReader eventReader = xmlInputFactory.createXMLEventReader(inputStream);
+
+            while (eventReader.hasNext()) {
+                XMLEvent xmlEvent = eventReader.nextEvent();
+                switch (xmlEvent.getEventType()) {
+                case XMLStreamConstants.START_ELEMENT:
+                    StartElement startElement = xmlEvent.asStartElement();
+                    loadIntermediateMuleObjects(startElement);
+                    break;
+
+                case XMLStreamConstants.CHARACTERS:
+                    break;
+
+                case XMLStreamConstants.END_ELEMENT:
+                    EndElement endElement = xmlEvent.asEndElement();
+                    checkFlowEnd(endElement);
+                    break;
+
+                default:
+                    break;
+                }
+            }
+
+        } catch (XMLStreamException e) {
+            logger.error(e.toString());
+        }
+    }
+
+    public InputStream getInputStream(String fileName) {
+        File file = new File(fileName);
+        FileInputStream fis = null;
+        try {
+            fis = new FileInputStream(file);
+        } catch (FileNotFoundException e) {
+            logger.error("Error");
```

Comment:
Can we add a more meaningful error log here?

Reply:
These were corrected in a later commit >> https://github.com/wso2/product-ei/pull/979/files#diff-ea4cf52b0ef514be466d6bf957833ba3L124